x-pack/plugins/monitoring/public/views/cluster/overview/index.html

```diff
@@ -1,3 +1,3 @@
-<monitoring-main name="overview" data-test-subj="clusterOverviewContainer">
+<monitoring-main name="overview" data-test-subj="clusterOverviewContainer" on-loaded="monitoringClusterOverview.init">
```

Comment:
You can also just use:
```javascript
ng-init="monitoringClusterOverview.init()"
```

Without any other custom logic.

But I think for what you're trying to do you can get away with:
 `ng-show="monitoringClusterOverview.showIfLoaded"`

Reply:
When I use that, the DOM element isn't available yet. I basically just want a hook to let me know when I can safely render into the DOM element. Any suggestions?

Reply:
Ah, then you can trigger it like this:
```javascript
      this.init = () => {
        $scope.$applyAsync(() => {
          console.log('...init', document.getElementById('monitoringClusterOverviewApp')); //will now be available
          $scope.$watch(...
        });
      }
```

Up to you though, your method seems to be working too, so it's fine imo